usvm-ts/src/main/kotlin/org/usvm/machine/expr/ExprUtil.kt

```diff
@@ -248,3 +248,16 @@ fun TsContext.checkReadingInRange(
         blockOnFalseState = { throwException("Index out of bounds: $index, length: $length") }
     )
 }
+
+fun TsContext.checkLengthBounds(
+    scope: TsStepScope,
+    length: UExpr<TsSizeSort>,
+    maxLength: Int,
+): Unit? {
+    // Check that length is non-negative and does not exceed `maxLength`.
+    val condition = mkAnd(
+        mkBvSignedGreaterOrEqualExpr(length, mkBv(0)),
+        mkBvSignedLessOrEqualExpr(length, mkBv(maxLength))
+    )
+    return scope.assert(condition)
```

Comment:
It is different conditions. Reading by a negative element should raise an exception, since it is probably an error. Reading by 1001 is not an error in all cases